pkg/kubelet/config/file_linux.go

```diff
@@ -151,3 +155,30 @@ func (s *sourceFile) consumeWatchEvent(e *watchEvent) error {
 	}
 	return nil
 }
+
+func modifyStaticPodForPinnedManagement(pod *v1.Pod) {
+	if !managed.IsPodManaged(pod) {
+		return
+	}
+	if pod.Annotations == nil {
+		pod.Annotations = make(map[string]string)
+	}
+	if pod.Labels == nil {
+		pod.Labels = make(map[string]string)
+	}
+	pod.Labels[managed.WorkloadsLabel] = ""
+	for _, container := range pod.Spec.Containers {
+		if container.Resources.Requests.Cpu().IsZero() {
+			continue
+		}
+		var resourceLimit string
+		if val, valid := container.Resources.Requests.Cpu().AsInt64(); valid {
+			resourceLimit = fmt.Sprintf("%v", managed.MilliCPUToShares(val*1000))
+		} else {
+			resourceLimit = fmt.Sprintf("%v", managed.MilliCPUToShares(int64(container.Resources.Requests.Cpu().AsApproximateFloat64()*1000)))
+		}
+		containerNameKey := fmt.Sprintf(managed.ContainerCpuAnnotationFormat, container.Name)
+		pod.Annotations[containerNameKey] = resourceLimit
+		delete(container.Resources.Requests, v1.ResourceName(v1.ResourceCPU))
+	}
```

Comment:
Also need to add workload.openshift.io/cores to the spec.

Reply:
fixed